SetupDataPkg/Library/ConfigKnobShimLib/ConfigKnobShimDxeLib/GoogleTest/ConfigKnobShimDxeLibGoogleTest.cpp

```diff
@@ -0,0 +1,176 @@
+/** @file ConfigKnobShimDxeLibGoogleTest.cpp
+  This is a sample to demonstrates the use of GoogleTest that supports host
+  execution environments.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/GoogleTestLib.h>
+#include <GoogleTest/Library/MockUefiRuntimeServicesTableLib.h>
+extern "C" {
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/ConfigKnobShimLib.h>
+}
+
+#define CONFIG_KNOB_GUID  {0x52d39693, 0x4f64, 0x4ee6, {0x81, 0xde, 0x45, 0x89, 0x37, 0x72, 0x78, 0x55}}
+
+using namespace testing;
+
+///////////////////////////////////////////////////////////////////////////////
+class GetConfigKnobOverrideFromVariableStorageTest : public Test
+{
+protected:
+MockUefiRuntimeServicesTableLib RtServicesMock;

```

Comment:
I think we should use strict mock for all mocked instances, so that the framework can call out missing mocked functions instead of returning default (0), resulting in false negative.

```suggestion
StrictMock<MockUefiRuntimeServicesTableLib> RtServicesMock;
```